1.3.c

```diff
@@ -1,121 +1,111 @@
-                break;
+                state = 1;
             }
-            isContinue = false;
-            break;
         }
-    }
-    return currentCharacter == '\0';
-}
+        case 3:
+        {
+            if (symbol == "_")
+            {
+                state = 1;
+            }
 
-bool tests()
-{
-    char testString1[] = "My_var001";
-    char testString2[] = "A";
-    char testString3[] = "_Afewew";
-    char testString4[] = "M_>_";
+        }
+        case 4:
+        {
+            if ((symbol >= 'A' && symbol <= 'Z') || (symbol >= 'a' && symbol <= 'z') || (symbol >= '0' && symbol <= '9') || (symbol == "_"))
+            {
+                state = 1;
+            }
+            else;
+            {
+                state = 2; // state can be 0,1. 2 is false.
 
-    return isCorrectString(testString1) &&
-        isCorrectString(testString2) &&
-        !isCorrectString(testString3) &&
-        !isCorrectString(testString4);
+            }
+        }
+     break;
+    }
+    return state;
 }
 
-#include <stdio.h>
-#include <stdbool.h>
-
-#define maximumSize 256
-
-bool isCorrectString(char* string)
+bool isCorrectString(char *string)
 {
     int state = 0;
-    char currentCharacter = -1;
-    int counter = 0;
-    bool isContinue = true;
-    while (isContinue && currentCharacter != '\0')
+    for (; *string != "/0"; string++)
     {
-        currentCharacter = string[counter++];
-        switch (state)
+        state = IncomingDataStatus(*string, IncomingDataStatus);
+        if (IncomingDataStatus == 2)
         {
-        case 0:
-            if ((currentCharacter >= 'A' && currentCharacter <= 'Z') || (currentCharacter >= 'a' && currentCharacter <= 'z'))
-            {
-                state = 1;
-                break;
-            }
             return false;
-        case 1:
-            if ((currentCharacter >= 'A' && currentCharacter <= 'Z') || (currentCharacter >= 'a' && currentCharacter <= 'z') ||
-                (currentCharacter >= '0' && currentCharacter <= '9') || (currentCharacter == '_'))
-            {
-                break;
-            }
-            isContinue = false;
-            break;
         }
     }
-    return currentCharacter == '\0';
+    return state == 1;
 }
 
-bool tests()
+bool trueTests()
 {
-    char testString1[] = "My_var001";
-    char testString2[] = "A";
-    char testString3[] = "_Afewew";
-    char testString4[] = "M_>_";
+    char test1[] = "MyNameIs_Agent007";
+    char test2[] = "F";
+    
 
-    return isCorrectString(testString1) &&
-        isCorrectString(testString2) &&
-        !isCorrectString(testString3) &&
-        !isCorrectString(testString4);
+    return isCorrectString(test1) &&
+        isCorrectString(test2);
 }
 
+bool falseTests()
+{
+    char test3[] = "_Rtfjkflf";
+    char test4[] = "A_+m-9";
+
+    return !isCorrectString(test3) &&
+        !isCorrectString(test4);
+       
+}
+ 
 int main()
 {
-    if (!tests())
+    if (trueTests == (false))

```

Comment:
```suggestion
    if (!trueTests())
```
Это не вызов тестов, это сравнение адресов. То же самое для негативных тестов
(-1)